### PR TITLE
`Integrated code lifecycle`: Use specified build job result paths

### PR DESCRIPTION
**Pull request by laurenzfb**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

Windfiles define the path(s) of the build results. For some programming languages (e.g. Java, Python), these defined paths contain wild cards which local CI previously was not able to handle and thus hardcoded paths were used. This is not ideal, as all languages should use the path(s) defined in the windfile.

### Description

This PR implements a bash command that is executed within the docker container and moves the results at the defined paths to a dedicated directory in the container. We are able to do this as bash can work with wild card paths. Artemis can then find all result files at a single location,

### Steps for Testing

Prerequisites:
**Test on TS3, TS6 or TS11**
- 1 Instructor

1. Create submission for exercises
   - Java Gradle
   - Java Gradle Sequential Test Runs
   - Java Gradle Static Code Analysis
   - Java Maven
   - Java MavenSequential Test Runs
   - Java Maven Static Code Analysis
   - Python
   - C
2. Verify the results

#### Performance Review
- [x] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Server
| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [LocalCIBuildJobExecutionService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5957/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.connectors.localci/LocalCIBuildJobExecutionService.html) | 75% | :white_check_mark: |
| [LocalCIContainerService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5957/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.connectors.localci/LocalCIContainerService.html) | 71% | :white_check_mark: |

Coverage further improved in #7861

## Summary by CodeRabbit

- **New Features**
  - Implemented a new method to efficiently move test results to a specified directory.

- **Bug Fixes**
  - Enhanced exception handling for test result path operations.
  - Fixed potential issues with handling multiple test result streams by utilizing a single stream.

- **Refactor**
  - Streamlined the process of retrieving test result paths.
  - Updated test suites to use a new, consistent results directory path.

- **Documentation**
  - No visible changes.

- **Style**
  - No visible changes.

- **Tests**
  - Improved test coverage with additional assertions and verifications.
  - Refactored tests to align with the new results directory structure.

- **Chores**
  - No visible changes.

- **Revert**
  - No visible changes.

### CODE REVIEW

File: src/main/resources/templates/aeolus/java/plain_maven_static.yaml [hunk 1]
```diff
@@ -15,3 +15,6 @@ actions:
         path: target/pmd.xml
       - name: pmd_cpd
         path: target/cpd.xml
+      - name: junit
```
Comment (reschandreas) [inline src/main/resources/templates/aeolus/java/plain_maven_static.yaml:18]: thanks for fixing this

**Review by reschandreas — APPROVED** [review]
tested locally, works as expected, code lgtm. thanks

File: src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIContainerService.java [hunk 2]
```diff
@@ -368,6 +382,12 @@ public void onComplete() {
         return buildLogEntries;
     }
 
+    private void checkPath(String path) {
+        if (path == null || path.contains("..") || !path.matches("[a-zA-Z0-9_*./-]+")) {
```
Comment (reschandreas) [inline src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCIContainerService.java:386]: here you could make use of the method `stripIllegalCharacters` in `StringUtil` to improve it and reuse code a bit :D

**Review by milljoniaer — APPROVED** [review]
tested in testing session 🚀

**Review by mateusmm01 — APPROVED** [review]
Code LGTM

**Pull request merged**